src/navigationService/index.js

```diff
@@ -27,6 +27,14 @@ export const ForceRerenderOnNavigation = ({ resetStateFunction }) => (
   />
 )
 
+export const ForceCleanUpOnScreenLeave = ({ cleanUpFunction }) => (
+  <NavigationEvents
+    onWillBlur={() => {
+      cleanUpFunction()
+    }}
```

Comment:
Is this lambda really necessary? If `cleanUpFunction` doesn't expect anything as parameters, then you can just pass that funciton as the event handler: `onWillBlur={cleanUpFunction}`